forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -237,3 +254,94 @@ impl WalletCommands {
         }
     }
 }
+
+fn formating_vars(balance_string: String, balance_exact: String) -> (String, String) {
+    if balance_exact.len() <= balance_string.len() {
+        let res = if balance_exact.ends_with(".0") {
+            //unfallible unwrap
+            balance_exact.strip_suffix(".0").unwrap().to_string()
+        } else {
+            balance_exact
+        };
+        (res, "".to_string())
+    } else {
+        //unfallible unwrap
+        let re = Regex::new(r"(\.0+|\.)$").unwrap();
+        let res = re.replace(balance_string.as_str(), "").to_string();
+        (res, "~".to_string())
+    }
+}
+
+fn format_balance_string(
+    mut balance_int: TokenAmount,
+    fixed_unit: &bool,
+    exact_balance: &bool,
+) -> String {
+    let mut unit = "FIL";
+    let (balance_string, symbol) = if *fixed_unit {
+        if *exact_balance {
+            formating_vars(format!("{balance_int}"), format!("{balance_int}"))
+        } else {
+            formating_vars(format!("{balance_int:.0}0"), format!("{balance_int}"))
+        }
+    } else {
+        let atto = balance_int.atto();
+        if *atto < BigInt::from(1000) {
+            unit = "atto FIL";
+            balance_int *= BigInt::from(1000000000000000000i64);
+        } else if *atto < BigInt::from(1000000) {
+            unit = "femto FIL";
+            balance_int *= BigInt::from(1000000000000000i64);
+        } else if *atto < BigInt::from(1000000000) {
+            unit = "pico FIL";
+            balance_int *= BigInt::from(1000000000000i64);
+        } else if *atto < BigInt::from(1000000000000i64) {
+            unit = "nano FIL";
+            balance_int *= BigInt::from(1000000000);
+        } else if *atto < BigInt::from(1000000000000000i64) {
+            unit = "micro FIL";
+            balance_int *= BigInt::from(1000000);
+        } else if *atto < BigInt::from(1000000000000000000i64) {
+            unit = "milli FIL";
+            balance_int *= BigInt::from(1000);
+        }
+        if *exact_balance {
+            formating_vars(format!("{balance_int}"), format!("{balance_int}"))
+        } else {
+            formating_vars(format!("{balance_int:.4}"), format!("{balance_int}"))
+        }
+    };
+    format!("{symbol}{balance_string} {unit}")
+}
+
+#[test]
+fn exact_balance_fixed_unit() {
+    assert_eq!(
+        format_balance_string(TokenAmount::from_atto(100), &true, &true,),
+        "0.0000000000000001 FIL"
+    );
+}
+
+#[test]
+fn not_exact_balance_fixed_unit() {
+    assert_eq!(
+        format_balance_string(TokenAmount::from_atto(100), &true, &false,),
+        "~0 FIL"
+    );
+}
+
+#[test]
+fn exact_balance_not_fixed_unit() {
+    assert_eq!(
+        format_balance_string(TokenAmount::from_atto(100), &false, &true,),
+        "100 atto FIL"
+    );
+}
+
+#[test]
+fn not_exact_balance_not_fixed_unit() {
+    assert_eq!(
+        format_balance_string(TokenAmount::from_atto(100), &false, &false,),
+        "100 atto FIL"
+    );
+}
```

Comment:
These tests are far from being exhaustive. You can definitely test 100% of your code.

Reply:
Added more